### PR TITLE
[testing] add `broadcasts_input` and verifies the behaviour for inplace_variant.

### LINKED ISSUE #55595: Verify that attempting to resize a tensor with an inplace operation throws a runtime error
Now that https://github.com/pytorch/pytorch/pull/53014 is in there is a mechanism used by test_variant_consistency_eager to filter sample inputs intended for use with the inplace variants of operators. This is intended to avoid sample inputs that would throw an error when used inplace. In particular, operations that try to resize a tensor inplace should throw an error:

```
a = torch.tensor((1,))
b = torch.tensor((2, 3, 4))
a.add_(b)
: RuntimeError: output with shape [1] doesn't match the broadcast shape [3]
```

However, https://github.com/pytorch/pytorch/issues/55589 shows that some operations do not error out in cases like this! I propose we extend test_variant_consistency_eager to not skip these inputs but verify the inplace operation throws an error when called with them.

cc @mruberry @VitalyFedyunin @walterddr @kshitij12345

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes #55595

* Add `broadcasts_input` attribute to SampleInput
* Update test_variant_consistency_eager to verify that sample with `broadcasts_input==True` and inplace variant raises an error.

### CODE REVIEW

File: test/test_ops.py
Comment (kshitij12345) [inline test/test_ops.py:292]: Received two different error messages, so haven't used `assertRaisesRegex`.

Message 1:
```
expand(CUDABoolType{[5, 5]}, size=[5]): the number of sizes provided (1) must be greater or equal to the number of dimensions in the tensor (2)
```

Message 2:
```
output with shape [5, 5] doesn't match the broadcast shape [5, 5, 5]
```

Comment (mruberry) [reply test/test_ops.py:292]: Not using assertRaisesRegex sounds good

**Review by mruberry — APPROVED** [review]
Cool! This might pick up a merge conflict. When it's about to be landed (after the internal CI and approval process) I'll ping so we can update it (if needed) and then land it immediately.

**Comment by kshitij12345** [comment]
@mruberry updated. Have tested locally. Lets wait for the CI to pass.

**Pull request closed without merging**
